addons/calendar/tests/test_calendar.py

```diff
@@ -381,6 +381,16 @@ def _test_emails_has_attachment(self, partners):
         # no more email should be sent
         _test_one_mail_per_attendee(self, partners)
 
+        # Check for attachment when public user creates an event ( simulates scheduling appointment from an incognito window)
+        dummy_partner = self.env['res.partner'].create({'name': 'Dummy User', 'email': 'Dummy@user.com'})
+        self.CalendarEvent.with_user(self.env.ref('base.public_user')).sudo().create({
+            'name': "publicUserEvent",
+            'partner_ids': [(6, False, dummy_partner.ids)],
+            'start': "2023-10-06 12:00:00",
+            'stop': "2023-10-06 13:00:00",
+        })
+        _test_emails_has_attachment(self, [dummy_partner])
```

Comment:
Seems like this does the search as sudo so I'm not sure we're actually testing what we want. Does it fail before the fix?

Reply:
It does fail.... but you got a point there. Cause I've noticed test is not exactly precisely replicating the flow from the event creation by a public user, but afaik it's not because we search as sudo, it's because we create the event as sudo

Reply:
While the test does fail/pass fine.... but you got a point there. It doesn't cover fully the desired scenario. Cause I've noticed that test is not exactly precisely replicating the flow from the event creation by a public user.
In the real flow the create_uid is set to the public user id=4
While in my test the create_uid is set to the to the creator of the odoo bot, which isn't public.